### DISCUSSION TITLE
Default strategy for index paths? e.g. composer/gitignore

### DISCUSSION THREAD

**Discussion by ging-dev** [phpactor/phpactor#2047] [Ideas]
This will help on some projects, for example Symfony, ignoring the `/var/cache`

**Comment by mamazu** [comment]
You can remove the pattern from the indexer manually:
https://phpactor.readthedocs.io/en/master/tips/performance.html?highlight=exclude_pattern#indexing

**Comment by dantleech** [comment]
Ignoring files in `.gitignore` isn't an option most of the time as this includes `vendor/` which we need to index.

a better strategy would be to use `composer.json` to determine what to index or not...

**Comment by dantleech** [comment]
Also it might be worth adding the indexer config for Symfony in the docs: https://phpactor.readthedocs.io/en/master/integrations/symfony.html

I use:

```
  "indexer.exclude_patterns": [
    "/vendor/**/Tests/**/*",
    "/vendor/**/tests/**/*",
    "/var/cache/**/*",
    "/vendor/composer/**/*"
  ]
```

using composer could also account for tests, at least in Symfony components ,e.g. from `Filesystem/composer.json`

```
    "autoload": {
        "psr-4": { "Symfony\\Component\\Filesystem\\": "" },
        "exclude-from-classmap": [
            "/Tests/"
        ]
    },
```

but could be a bit tricky.

**Reply by mamazu** [reply]
One thing that also should be taken into account: In newer versions of Symfony >= 5 you can have php configuration for bundles and the configuration classes are generated under `/var/cache/dev/Symfony/Config` so the composer approach would probably be the best idea.
